File: app/src/main/java/droiddevelopers254/droidconke/adapters/AgendaAdapter.java
package droiddevelopers254.droidconke.adapters;

import android.content.Context;
import android.graphics.Color;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.bumptech.glide.Glide;
import com.bumptech.glide.load.engine.DiskCacheStrategy;
import com.bumptech.glide.load.resource.drawable.DrawableTransitionOptions;
import com.bumptech.glide.request.RequestOptions;

import java.util.List;

import droiddevelopers254.droidconke.R;
import droiddevelopers254.droidconke.models.AgendaModel;

public class AgendaAdapter extends RecyclerView.Adapter<AgendaAdapter.MyViewHolder> {
    private List<AgendaModel> agendaModelList;
    private Context context;

    public AgendaAdapter(List<AgendaModel> agendaModelList, Context context){
        this.agendaModelList = agendaModelList;
        this.context=context;
    }
    public class MyViewHolder extends RecyclerView.ViewHolder {
        TextView agendaTitleText,agendaTimelineText;
        ImageView agendaImg;
        LinearLayout agendaLinear;
        public MyViewHolder(@NonNull View itemView) {
            super(itemView);
            agendaTitleText=itemView.findViewById(R.id.agendaTitleText);
            agendaTimelineText=itemView.findViewById(R.id.agendaTimelineText);
            agendaImg=itemView.findViewById(R.id.agendaImg);
            agendaLinear=itemView.findViewById(R.id.agendaLinear);
        }
    }

    @NonNull
    @Override
    public MyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View itemView = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.agenda_details, parent, false);
        return new MyViewHolder(itemView);
    }

    @Override
    public void onBindViewHolder(@NonNull MyViewHolder holder, int position) {
        AgendaModel agendaModel = agendaModelList.get(position);
        holder.agendaTitleText.setText(agendaModel.getTitle());
        holder.agendaTimelineText.setText(agendaModel.getTime());
        holder.agendaLinear.setBackgroundColor(Color.parseColor(agendaModel.getBackground_color()));

        //load speaker profile image
        Glide.with(context).load(agendaModel.getIconUrl())
                .thumbnail(Glide.with(context).load(agendaModel.getIconUrl()))
                .transition(new DrawableTransitionOptions()
                        .crossFade())
                .apply(new RequestOptions()
                        .centerCrop()
                        .diskCacheStrategy(DiskCacheStrategy.ALL))
                .into(holder.agendaImg);
    }

    @Override
    public int getItemCount() {
        return agendaModelList.size();
    }


}

File: app/src/main/java/droiddevelopers254/droidconke/adapters/AgendaAdapter.kt
package droiddevelopers254.droidconke.adapters

import android.content.Context
import android.graphics.Color
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import droiddevelopers254.droidconke.R
import droiddevelopers254.droidconke.models.AgendaModel
import kotlinx.android.synthetic.main.agenda_details.view.*

class AgendaAdapter(private val agendaModelList: List<AgendaModel>, private val context: Context) : RecyclerView.Adapter<AgendaAdapter.MyViewHolder>() {
    inner class MyViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        private val agendaTitleText = itemView.agendaTitleText
        private val agendaTimelineText = itemView.agendaTimelineText
        private val agendaImg =itemView.agendaImg
        private val agendaLinear = itemView.agendaLinear

        fun bindAgendas(agendaModel: AgendaModel){
            with(agendaModel){
                //TODO add logic for changing agenda icon
                agendaTitleText.text = title
                agendaTimelineText.text = time
                agendaLinear.setBackgroundColor(Color.parseColor(background_color))
            }
        }

    }
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        val itemView = LayoutInflater.from(parent.context)
                .inflate(R.layout.agenda_details, parent, false)
        return MyViewHolder(itemView)
    }

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
       holder.bindAgendas(agendaModelList[position])
    }
    override fun getItemCount(): Int {
        return agendaModelList.size
    }


}
